addons/website_sale_stock_wishlist/static/src/js/website_sale.js

```diff
@@ -6,30 +6,33 @@ import ajax from "web.ajax";
 import { qweb as QWeb } from "web.core";
 
 const loadXml = async () => {
-  return ajax.loadXML(
-    "/website_sale_stock_wishlist/static/src/xml/product_availability.xml",
-    QWeb
-  );
+    return ajax.loadXML('/website_sale_stock_wishlist/static/src/xml/product_availability.xml', QWeb);
 };
 
 publicWidget.registry.WebsiteSale.include({
-  /**
-   * It will display additional info messages regarding the select product's stock and the wishlist.
-   * @override
-   */
-  _onChangeCombination: function (ev, $parent, combination) {
-    this._super.apply(this, arguments);
-    loadXml().then(function () {
-      if ($('.o_add_wishlist_dyn').length) {
-        var $message = $(
-          QWeb.render(
-            "website_sale_stock_wishlist.product_availability",
-            combination
-          )
-        );
-        $message.appendTo($("div.availability_messages"));
-      }
-    });
-  },
-});
 
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Displays additional info messages regarding the product's
+     * stock and the wishlist.
+     *
+     * @override
+     */
+    _onChangeCombination: async function (ev, $parent, combination) {
+        this._super(...arguments);
+        const self = this;
+        loadXml().then(() => {
+            if (self.el.querySelector('.o_add_wishlist_dyn')) {
+                const messageTemplate = document.createElement('template');
+                messageTemplate.innerHTML = QWeb.render('website_sale_stock_wishlist.product_availability', combination);
+                const messageNode = messageTemplate.content.firstElementChild;
```

Comment:
[Not supported on IE11](https://caniuse.com/?search=template.content) (which is fine for editor stuff but for website visitor stuff, I think we still consider to support it (and when easy alternatives are available, there is no reason not to))

In this case, this is more than an alternative, there is a more readable, faster and shorter version unless I am missing something:
```javascript
const messageEl = this.el.querySelector('...');
messageEl.insertAdjacentHTML('beforeend', QWeb.render('...'));
```